portal-api/api-tests/deal-versioning.api-test.js

```diff
@@ -0,0 +1,236 @@
+  let aMaker;
+  let anAdmin;
+  let testUsers;
+
+  beforeAll(async () => {
+    testUsers = await testUserCache.initialise(app);
+    aMaker = testUsers().withRole(MAKER).one();
+    anAdmin = testUsers().withRole(ADMIN).one();
+  });
+
+  beforeEach(async () => {
+    await databaseHelper.wipe([MONGO_DB_COLLECTIONS.DEALS, MONGO_DB_COLLECTIONS.FACILITIES, MONGO_DB_COLLECTIONS.ELIGIBILITY_CRITERIA]);
+    await as(anAdmin).post(mockEligibilityCriteriaVersion).to('/v1/gef/eligibility-criteria');
+  });
+
+  describe(`when GEF_DEAL_VERSION set to '0'`, () => {
+    beforeAll(() => {
+      process.env.GEF_DEAL_VERSION = '0';
+    });
+
+    afterAll(() => {
+      process.env = originalEnv;
+    });
+
+    it('returns the expected version 0 application upon creation', async () => {
+      const { body } = await as(aMaker).post(generateVersion0ApplicationToSubmit()).to(gefApplicationsUrl);
+
+      expect(body).toEqual(expectMongoId(generateVersion0ApplicationResponse(aMaker._id)));
+    });
+
+    it('returns the expected version 0 facility on creation', async () => {
+      const { body: dealBody } = await as(aMaker).post(generateVersion0ApplicationToSubmit()).to(gefApplicationsUrl);
+      const { body } = await as(aMaker).post({ dealId: dealBody._id, type: FACILITY_TYPE.CASH, hasBeenIssued: false }).to(gefFacilitiesUrl);
+
+      expect(body).toEqual(generateVersion0Facility(dealBody._id, aMaker._id));
+    });
+
+    it('returns 400 when updating isUsingFacilityEndDate on a version 0 facility', async () => {
+      const { body: dealBody } = await as(aMaker).post(generateVersion0ApplicationToSubmit()).to(gefApplicationsUrl);
+      const { body: facilityBody } = await as(aMaker).post({ dealId: dealBody._id, type: FACILITY_TYPE.CASH, hasBeenIssued: false }).to(gefFacilitiesUrl);
+      const { status, body } = await as(aMaker).put({ isUsingFacilityEndDate: true }).to(`${gefFacilitiesUrl}/${facilityBody.details._id}`);
+
+      expect(body).toEqual({ status: 400, message: 'Cannot add facility end date to deal version 0' });
+      expect(status).toBe(400);
+    });
+
+    it('returns 400 when creating a version 0 facility with isUsingFacilityEndDate', async () => {
+      const { body: dealBody } = await as(aMaker).post(generateVersion0ApplicationToSubmit()).to(gefApplicationsUrl);
+      const { status, body } = await as(aMaker)
+        .post({ dealId: dealBody._id, type: FACILITY_TYPE.CASH, hasBeenIssued: false, isUsingFacilityEndDate: true })
+        .to(gefFacilitiesUrl);
+
+      expect(body).toEqual({ status: 400, message: 'Cannot add facility end date to deal version 0' });
+      expect(status).toBe(400);
+    });
+  });
+
+  describe(`when GEF_DEAL_VERSION set to '1'`, () => {
+    beforeAll(() => {
+      process.env.GEF_DEAL_VERSION = '1';
+    });
+
+    afterAll(() => {
+      process.env = originalEnv;
+    });
+
+    it('returns 200 when updating isUsingFacilityEndDate on a version 0 facility', async () => {
```

Comment:
We might have checked this elsewhere, and this might not be the right place for it but do we want to check we can't post a non boolean isUsingFacilityEndDate to an update/create request

Reply:
Oops, had added tests below but forgot to change the name